File: 118. Pascal's Triangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> result = new ArrayList<List<Integer>>();
        if (numRows == 1){
            result.add(new ArrayList<>());
            result.get(0).add(1);
            return result;
        }
        
        List<List<Integer>> prev = generate(numRows-1);
        result.addAll(prev);
        List<Integer> last = result.get(result.size()-1);
        int i = 0;
        int j = 0;
        List<Integer> newL = new ArrayList<Integer> ();
        while (i <= last.size()){
            if (i == 0 ){
                newL.add(last.get(i));
            }
            else if (i == last.size()){
                newL.add(last.get(i-1));
            }
            
            else{
                newL.add(last.get(i)+last.get(j));
                j++;
            }
            i++;
               
        }
        result.add(newL);
        return result;
    }
}

File: 118. Pascal's Triangle.py
class Solution(object):
    def generate(self, numRows):
        """
        :type numRows: int
        :rtype: List[List[int]]
        """
        
        result = []
        if numRows == 1:
            result.append([1])
            return result
        
        else:
            temp = self.generate(numRows-1)
            result.extend(temp)
            last_row = temp[-1]
            new = []
            i = 0
            j= 0
            
            while i <= len(last_row):
                if i == 0:
                    new.append(last_row[i])
                elif i == len(last_row):

                    new.append(last_row[-1])
                elif i > j:
                    new.append(last_row[i] + last_row[j])
                    j += 1
                
                
                i += 1
            result.append(new)
            return result
